R/aldex.R

```diff
@@ -93,74 +86,30 @@ aldex <- function(Y, X, data=NULL, nsample=2000,  scale=NULL,
     formula <- NULL
   }
 
-  ## perform streaming 
   out <- list()
-  nsample.remaining <- nsample
   iter <- 1
-  if (stream) {
-    nsample.local <- floor(streamsize*100000/(N*D*8))
-    if (nsample.local < 1) stop("streamsize too small")
-  } else {
-    nsample.local <- nsample
+  chunk.sizes <- aldex.getchunksizes(Y, X, nsample, streamsize)
+  for(chunk.size in chunk.sizes) {
+    chunk <- aldex.chunk(Y, X, chunk.size, scale, scale.args,
```

Comment:
I am ok with splitting this functionality off but want to review again once there are fewer other changes. When its easier to really see what was changed and what was not. Just make sure this has the same memory footprint as my code. These matricies can easily get to 10Gb each so a single needless copy is a no-go in my book.